### PR TITLE
[FIX] account: set default value on journal_id in accrued order wizard

### PR DESCRIPTION
**Pull request by ghost**
When creating an accrued revenue entry, the same journal will always be selected by the model (by default: Miscellaneous Operations) even after selecting and confirming another journal in the view form. This is due to the computed field that always take the same journal. Converting the computed function to a default function resolve the issue.

related PR: #109058

opw-3076067

### CODE REVIEW

**Review by nle-odoo — APPROVED** [review]
this would remove the computation of the record based on company_id which I don't think is something we want. Also this is causing an error in test since this leads to journal_id being empty but there is a `required=True` parameter on the field.

instead of this you should be able to just add `stored=True` to have a stored computed field

File: addons/account/wizard/accrued_orders.py [hunk 1]
```diff
@@ -23,10 +23,13 @@ def _get_default_company(self):
         orders = self.env[self._context['active_model']].browse(self._context['active_ids'])
         return orders and orders[0].company_id.id
 
+    def _get_default_journal(self):
+        return self.env['account.journal'].search([('company_id', '=', self.env.company.id), ('type', '=', 'general')], limit=1)
+
```
Comment (william-andre) [inline addons/account/wizard/accrued_orders.py:28]:
```suggestion
```

File: addons/account/wizard/accrued_orders.py [hunk 2]
```diff
@@ -23,10 +23,13 @@ def _get_default_company(self):
...
     company_id = fields.Many2one('res.company', default=_get_default_company)
     journal_id = fields.Many2one(
         comodel_name='account.journal',
-        compute='_compute_journal_id',
+        default=_get_default_journal,
```
Comment (william-andre) [inline addons/account/wizard/accrued_orders.py:32]:
```suggestion
        compute='_compute_journal_id", store=True, readonly=False,
```

File: addons/account/wizard/accrued_orders.py [hunk 3]
```diff
@@ -23,10 +23,13 @@ def _get_default_company(self):
...
         domain="[('type', '=', 'general'), ('company_id', '=', company_id)]",
         readonly=False,
```
Comment (william-andre) [inline addons/account/wizard/accrued_orders.py:34]:
```suggestion
```

**Review by william-andre — CHANGES_REQUESTED** [review]
Avoid `default` whenever possible.
You shouldn't loo, at `self.env.company`, but rather at `record.company_id`

**Comment by joda-odoo** [comment]
> Avoid `default` whenever possible. 
A test in runbot didn't like the fact that the field was null be default, that's why I've added a default value.

File: addons/account/wizard/accrued_orders.py (hunk 2 above)
Comment (joda-odoo) [reply addons/account/wizard/accrued_orders.py:32]: I've tried like this: 
```python
    journal_id = fields.Many2one(
        comodel_name='account.journal',
        compute='_compute_journal_id',
        domain="[('type', '=', 'general'), ('company_id', '=', company_id)]",
        readonly=False,
        store=True,
        required=True,
        check_company=True,
        string='Journal',
    )
```

And it doesn't work. It keep coming back to the `Misc journal`

**Comment by joda-odoo** [comment]
> Avoid `default` whenever possible. 

One of the test failed on runbot, because it was checking the the field wasn't null ([the runbot in question](https://runbot.odoo.com/runbot/build/22630193))

File: addons/account/wizard/accrued_orders.py (hunk 2 above)
Comment (william-andre) [reply addons/account/wizard/accrued_orders.py:32]: Then just precompute the field
```suggestion
        compute='_compute_journal_id", store=True, readonly=False, precompute=True,
```

**Comment by william-andre** [comment]
@robodoo r+

**Comment by Levizar** [comment]
Hello @joda-odoo , this pr is rotting because the CI randomly failed. 
I restarted the failing build.
In the future, could you follow up on your r+ed pr as we have too many of them to watch out on that.

**Comment by joda-odoo** [comment]
Oh sorry I completly forgot about it. Thanks for reminding me about it :)

**Pull request closed without merging**